dev-docs/modules/consentManagement.md

```diff
@@ -0,0 +1,179 @@
+top_nav_section: dev_docs
+nav_section: modules
+module_code : consentManagement
+display_name : GDPR ConsentManagement
+---
+
+<div class="bs-docs-section" markdown="1">
+
+## PENDING RELEASE
+This module is still considered under development, but the core logic and integration steps documented here will very likely remain as described.
+
+# GDPR ConsentManagement Module
+{:.no_toc}
+
+Designed to support the EU General Data Protection Regulation ([GDPR](https://www.eugdpr.org/)), this module works with supported Consent Management Platforms (CMPs) to fetch an encoded string representing the user's consent choices and make it available for adapters to consume and process.
+
+This module will perform its tasks with the CMP prior to the auction starting.  A rough synopsis of this interaction process would be:
+
+1. Fetch the user's encoded consent string from the CMP.
+2. If this request doesn't return a string, we assume this is a new-user and is undergoing the consent process.  We will wait for a signal from the CMP that the consent has finished and perform the lookup again.
+3. With a valid consent string, we will incorporate this data into the auction objects (for adapters to collect) and then allow the auction to proceed.
+
+There are timeout settings in place in the module to permit this interaction with the CMP a specified length of time to operate before it's unacceptable or assumed an issue has occurred.  
+
+When either this timeout occurs or if an error from the CMP is thrown, one of two options are taken; either:
+
+1. The auction is canceled outright.
+2. The auction proceeds without the user's consent information.  
+
+Though these options are mutually exclusive, they are configurable by the publisher via the site's implementation of the prebid code (see further below for details) so that they can be used in the proper scenarios for that site/audience.
+
+
+## Page integration
+
+To utilize this module, a separate CMP needs to be implemented onto the site to interact with the user and obtain their consent choices.  
+
+The actual implementation details of this CMP are not covered by this page; any questions on that implemenation should be referred to the CMP in question.  However, we would recommend to have the CMP's code located before the prebid code in the head of the page, in order to ensure their framework is implemented before the prebid code starts to execute.
+
+The module currently supports the following CMPs:
+
+* AppNexus (appnexus)
+
+Once the CMP is implemented, simply include the module in your build and add a consentManagement object in the setConfig() call.  Adapters that support this feature will be able to retrieve the consent information and incorporate it in their requests.
+
+{: .table .table-bordered .table-striped }
+| Param | Type | Description | Example |
+| --- | --- | --- | --- |
+| cmp | `string` | The ID for the CMP in use on the page.  Default is `'appnexus'` | `'appnexus'` |
```

Comment:
Related to the previous comment, believe this would be more like cmpApi, with the value being 'iab'.

Reply:
I'll make this change (note - there will be subsequent updates in the main PR as a result).